adc/ad7779/imx6ull/dma.c

```diff
@@ -98,16 +82,67 @@ static int sdma_configure(size_t size, size_t count, addr_t *phys_addr)
 	cfg.priority = SDMA_CHANNEL_PRIORITY_MIN + 1;
 	sdma_channel_configure(&sdma_common.sdma, &cfg);
 
-	sdma_common.buffer.size = aligned_size;
+	return 0;
+}
+
+
+static int sdma_init(size_t size, size_t count, addr_t *phys_addr)
+{
+	int res;
+	unsigned tries = 25;
+	addr_t bd_paddr;
+
+	size_t aligned_size = size;
+	if (aligned_size % _PAGE_SIZE) {
+		aligned_size += _PAGE_SIZE - aligned_size % _PAGE_SIZE;
+	}
+
+	while ((res = sdma_open(&sdma_common.sdma, SDMA_DEVICE_FILE_NAME)) < 0) {
```

Comment:
```suggestion
	res = sdma_open(&sdma_common.sdma, SDMA_DEVICE_FILE_NAME);
	while (res < 0) {
```

Reply:
This one is not good idea.

Reply:
I think @agkaminski  meant something similar to this:

```suggestion
	for (;;) {
		res = sdma_open(&sdma_common.sdma, SDMA_DEVICE_FILE_NAME);
		if (res >= 0) {
			break;
		}

		usleep(100 * 1000);
		if (--tries == 0) {
			log_error("failed to open SDMA device file (%s)", SDMA_DEVICE_FILE_NAME);
			return -1;
		}
	}
```

Reply:
`res` is unused, so then
```suggestion
	while (sdma_open(&sdma_common.sdma, SDMA_DEVICE_FILE_NAME) < 0) {
```